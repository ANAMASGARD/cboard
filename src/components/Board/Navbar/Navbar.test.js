import React from 'react';
import { shallow, mount } from 'enzyme';

import NavBar from './NavBar';
import { shallowMatchSnapshot } from '../../../common/test_utils';

const COMPONENT_PROPS = {

  intl: {},
  board: {
    email: 'qa@qa.com',
    isPublic: true
  },
  userData: { email: 'qa@qa.com' },
  onLockNotify: jest.fn(),
  className: 'string',
  title: 'string',
  disabled: false,
  isLocked: false,
  onBackClick: jest.fn(),
  onLockClick: jest.fn(),
  isScannerActive: false,
  onDeactivateScannerClick: jest.fn()
};

describe('NavBar tests', () => {
  global.navigator = Object.create(navigator);
  Object.defineProperty(navigator, 'share', {
    value: () => { }
  });
  it('renders without crashing unlocked', () => {
    const wrapper = shallow(<NavBar isLocked={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders without crashing locked', () => {
    const wrapper = shallow(<NavBar isLocked={true} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('default renderer', () => {
    shallowMatchSnapshot(<NavBar {...COMPONENT_PROPS} />);
  });
  test('check share', () => {
    const wrapper = shallow(<NavBar {...COMPONENT_PROPS} />);
  });
});



