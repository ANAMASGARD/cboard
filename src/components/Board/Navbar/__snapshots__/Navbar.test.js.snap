// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests default renderer 1`] = `
<Navbar
  board={
    Object {
      "email": "qa@qa.com",
      "isPublic": true,
    }
  }
  className="string"
  disabled={false}
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  isLocked={false}
  isScannerActive={false}
  onBackClick={[MockFunction]}
  onDeactivateScannerClick={[MockFunction]}
  onLockClick={[MockFunction]}
  onLockNotify={[MockFunction]}
  title="string"
  userData={
    Object {
      "email": "qa@qa.com",
    }
  }
/>
`;

exports[`NavBar tests renders without crashing locked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    isLocked={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "intl": Object {
        "formatMessage": [Function],
      },
      "isLocked": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "intl": Object {
          "formatMessage": [Function],
        },
        "isLocked": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`NavBar tests renders without crashing unlocked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    isLocked={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "intl": Object {
        "formatMessage": [Function],
      },
      "isLocked": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "intl": Object {
          "formatMessage": [Function],
        },
        "isLocked": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
